File: phone/ios/Sources/DashboardModels.swift
import Foundation

// dashboard/v1 snapshot şemasının Codable karşılığı.
// CodexBridge host (Windows) ve codexbar serve (macOS/Linux) aynı şemayı üretir.

struct DashboardSnapshot: Codable {
    var schemaVersion: Int = 1
    var generatedAt: String
    var staleAfterSeconds: Int = 180
    var host: HostInfo
    var providers: [ProviderRow] = []
}

struct HostInfo: Codable {
    var codexBarVersion: String?
    var refreshIntervalSeconds: Int = 0
}

struct ProviderRow: Codable {
    var id: String
    var name: String
    var enabled: Bool = true
    var source: String?
    var status: ProviderStatus?
    var identity: ProviderIdentity?
    var windows: [RateWindow] = []
    var credits: CreditInfo?
    var cost: CostInfo?
    var display: DisplayHints?
    var error: ProviderError?
    var updatedAt: String?
}

struct ProviderStatus: Codable {
    var level: String = "unknown"
    var label: String?
    var updatedAt: String?
}

struct ProviderIdentity: Codable {
    var accountEmail: String?
    var plan: String?
}

struct RateWindow: Codable {
    var kind: String
    var label: String?
    var usedPercent: Double?
    var remainingPercent: Double?
    var resetAt: String?
}

struct CreditInfo: Codable {
    var remaining: Double?
    var unit: String?
}

struct CostInfo: Codable {
    var todayUSD: Double?
    var last30DaysUSD: Double?
}

struct DisplayHints: Codable {
    var accentColor: String?
    var sortKey: Int = 0
    var priority: String?
}

struct ProviderError: Codable {
    var code: String?
    var message: String?
}

File: phone/android/src/main/java/com/codexbridge/widget/DashboardModels.kt
package com.codexbridge.widget

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

/**
 * dashboard/v1 snapshot şemasının kotlinx.serialization karşılığı.
 * CodexBridge host (Windows) ve codexbar serve (macOS/Linux) aynı şemayı üretir.
 */
@Serializable
data class DashboardSnapshot(
    val schemaVersion: Int = 1,
    val generatedAt: String,
    val staleAfterSeconds: Int = 180,
    val host: HostInfo,
    val providers: List<ProviderRow> = emptyList(),
)

@Serializable
data class HostInfo(
    val codexBarVersion: String? = null,
    val refreshIntervalSeconds: Int = 0,
)

@Serializable
data class ProviderRow(
    val id: String,
    val name: String,
    val enabled: Boolean = true,
    val source: String? = null,
    val status: ProviderStatus? = null,
    val identity: ProviderIdentity? = null,
    val windows: List<RateWindow> = emptyList(),
    val credits: CreditInfo? = null,
    val cost: CostInfo? = null,
    val display: DisplayHints? = null,
    val error: ProviderError? = null,
    val updatedAt: String? = null,
)

@Serializable
data class ProviderStatus(val level: String = "unknown", val label: String? = null, val updatedAt: String? = null)

@Serializable
data class ProviderIdentity(val accountEmail: String? = null, val plan: String? = null)

@Serializable
data class RateWindow(
    val kind: String,
    val label: String? = null,
    val usedPercent: Double? = null,
    val remainingPercent: Double? = null,
    val resetAt: String? = null,
)

@Serializable
data class CreditInfo(val remaining: Double? = null, val unit: String? = null)

@Serializable
data class CostInfo(
    @SerialName("todayUSD") val todayUsd: Double? = null,
    @SerialName("last30DaysUSD") val last30DaysUsd: Double? = null,
)

@Serializable
data class DisplayHints(val accentColor: String? = null, val sortKey: Int = 0, val priority: String? = null)

@Serializable
data class ProviderError(val code: String? = null, val message: String? = null)
